src/astro/databases/base.py

```diff
@@ -250,3 +281,66 @@ def export_table_to_file(
 
         df = self.export_table_to_pandas_dataframe(source_table)
         target_file.create_from_dataframe(df)
+
+    # ---------------------------------------------------------
+    # Schema Management
+    # ---------------------------------------------------------
+
+    def create_schema_if_needed(self, schema):
+        """
+        This function checks if the expected schema exists in the database. If the schema does not exist,
+        it will attempt to create it.
+
+        :param schema: DB Schema - a namespace that contains named objects like (tables, functions, etc)
+        """
+        if not schema:
+            return
+        if not self.schema_exists(schema):
+            statement = self._create_schema_statement.format(schema)
+            self.run_sql(statement)
+
+    def schema_exists(self, schema):
+        """
+        Checks if a schema exists in the database
+
+        :param schema: DB Schema - a namespace that contains named objects like (tables, functions, etc)
+        """
+        raise NotImplementedError
+
+    # ---------------------------------------------------------
+    # Context & Template Rendering methods (Transformations)
+    # ---------------------------------------------------------
+
+    def add_templates_to_context(
+        self, parameters: Dict, context: Dict
+    ) -> Dict:  # skipcq
+        """
+        When running functions through the `aql.transform` and `aql.render` functions, we need to add
+        the parameters given to the SQL statement to the Airflow context dictionary. This is how we can
+        then use jinja to render those parameters into the SQL function when users use the {{}} syntax
+        (e.g. "SELECT * FROM {{input_table}}").
+
+        With this system we should handle Table objects differently from other variables. Since we will later
+        pass the parameter dictionary into SQLAlchemy, the safest (From a security standpoint) default is to use
+        a `:variable` syntax. This syntax will ensure that SQLAlchemy treats the value as an unsafe template. With
+        Table objects, however, we have to give a raw value or the query will not work. Because of this we recommend
+        looking into the documentation of your database and seeing what best practices exist (e.g. Identifier wrappers
+        in snowflake).
+
+        :param parameters: A Dict of SQL key-value parameters
+        :param context: Airflow Context dictionary
+        :return: Dictionary with values with Table type replaced with the table name
+        """
+        for k, v in parameters.items():
+            if isinstance(v, Table):
+                context[k] = self.get_table_qualified_name(v)
+            else:
+                context[k] = ":" + k
+        return context
+
+    def process_sql_parameters(self, parameters: Dict):
```

Comment:
Concern raised in:
https://github.com/astronomer/astro-sdk/pull/353/files#r875922817